README.md

```diff
@@ -1,289 +1,610 @@
-* `<service-id>` is the `id` field of the corresponding `service` entry in the `ServiceBinding` CR.  If the `id` field is not present, the `name` field is used instead.  The `<service-id>` path **MUST** be unique between the services bound to a particular application.
-* `<persisted_secret>` represents a set of files where the filename is a Secret key and the file contents is the corresponding value of that key.
+## Binding Values as Environment Variables
 
-Example:  `/platform/bindings/my-nosql-db/secret/MONGODB_HOST`
+Many applications, especially initially, will not be able to consume Service Bindings as defined by the Application Projection section directly since many of these applications assume that configuration will be exposed via environment variables.  Users should have a way of describing how they would like environment variables containing the values from bound secrets mapped into their applications.  This specification is described as an extension to the [Service Binding](#service-binding) specification and assumes full compatibility with it.
 
-In addition to the secret, implementations of this specification **MUST** mount, if available, metadata about the bindings in the following location:
+A Service Binding Resource **MAY** define a `.spec.env` which is an array of `EnvVar`.  The value of an entry in this array **MAY** contain zero or more tokens beginning with `((`, ending with `))`, and encapsulating a binding `Secret` key name.  The value of this `Secret` entry **MUST** be substituted into the original `value` string, replacing the token.  Once all tokens have been substituted, the new `value` **MUST** be configured as an environment variable on the resource represented by `application`.
 
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name:         # string
+spec:
+  name:         # string, optional, default: .metadata.name
+
+  application:  # PodSpec-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  service:      # Provisioned Service-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  env:          # []EnvVar, optional
+  - name:       # string
+    value:      # string
+  ...
 ```
-$SERVICE_BINDINGS_ROOT/<service-id>/metadata/<persisted_metadata>
+
+### Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name: online-banking-to-account-service
+spec:
+  name: account-service
+
+  application:
+    apiVersion: apps/v1
+    kind:       Deployment
+    name:       online-banking
+
+  service:
+    apiVersion: com.example/v1alpha1
+    kind:       AccountService
+    name:       prod-account-service
+
+  env:
```

Comment:
Apologies, it's not clear to me why we need `env`. Is it because by default, nothing is exposed to the app as  environment variables - since everything is mounted to the volume path?

Reply:
yup.  `env` gives you a way to expose any of the items inside the final Secret (which will be mounted) as an environment variable.  And while doing that you can pick a new name - just for the env var, as the original name is still used for the mount.

If you want to rename something in the final Secret / mount, you would use `mappings`.

Reply:
@arthurdm hits the nail on the head, how to map things and how to expose them via env-vars are orthogonal to one another.